apis/python/src/tiledbsc/annotation_dataframe.py

```diff
@@ -65,6 +65,22 @@ def keys(self) -> List[str]:
         """
         return self.get_attr_names()
 
+    # ----------------------------------------------------------------
+    def attribute_filter(self, query_string, col_names_to_keep):
```

Comment:
Should we have a more generic function? What if a user wants to filter on attributes and provide query ranges?

Reply:
@Shelnutt2 good point -- I should have marked this as a 'v1' with a `TODO` -- definitely room for growth.

Reply:
https://github.com/single-cell-data/TileDB-SingleCell/pull/103/commits/34d8a0c9eb6b7383484aa5e08a2b8e4e3ca6b043